python/cuml/neighbors/kneighbors_classifier.pyx

```diff
@@ -0,0 +1,257 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+from cuml.neighbors.nearest_neighbors import NearestNeighbors
+
+from cuml.utils import get_cudf_column_ptr, get_dev_array_ptr, \
+    input_to_dev_array, zeros, row_matrix
+
+import numpy as np
+
+from cuml.metrics import accuracy_score
+
+import cudf
+
+from cython.operator cimport dereference as deref
+
+from cuml.common.handle cimport cumlHandle
+from libcpp.vector cimport vector
+
+
+from libcpp cimport bool
+from libcpp.memory cimport shared_ptr
+
+import rmm
+from libc.stdlib cimport malloc, free
+
+from libc.stdint cimport uintptr_t, int64_t
+from libc.stdlib cimport calloc, malloc, free
+
+from numba import cuda
+import rmm
+
+cimport cuml.common.handle
+cimport cuml.common.cuda
+
+cdef extern from "cuml/neighbors/knn.hpp" namespace "ML":
+
+    void knn_classify(
+        cumlHandle &handle,
+        int* out,
+        int64_t *knn_indices,
+        vector[int*] &y,
+        size_t n_samples,
+        int k
+    ) except +
+
+    void knn_class_proba(
+        cumlHandle &handle,
+        vector[float*] &out,
+        int64_t *knn_indices,
+        vector[int*] &y,
+        size_t n_samples,
+        int k
+    ) except +
+
+
+class KNeighborsClassifier(NearestNeighbors):
+
+    def __init__(self, weights="uniform", **kwargs):
+        """
+
+        :param kwargs:
+        """
+        super(KNeighborsClassifier, self).__init__(**kwargs)
+        self.y = None
+        self.weights = weights
+
+        if weights != "uniform":
+            raise ValueError("Only uniform weighting strategy is "
+                             "supported currently.")
+
+    def fit(self, X, y, convert_dtype=True):
+        """
+        Fit a k-nearest neighbors classifier model.
+        :param X:
+        :param y:
+        :param convert_dtype:
+        :return:
+        """
+        super(KNeighborsClassifier, self).fit(X, convert_dtype)
+
+        self.y, _, _, _, _ = \
+            input_to_dev_array(y, order='F', check_dtype=np.int32,
+                               convert_to_dtype=(np.int32
+                                                 if convert_dtype
+                                                 else None))
+
+        self.handle.sync()
+
+    def predict(self, X, convert_dtype=True):
+        """
+        Use the trained k-nearest neighbors classifier to
+        predict the labels for X
+        :param X:
+        :param convert_dtype:
+        :return:
+        """
+        return self._classify(X, convert_dtype)
```

Comment:
why the "private" `_classify` mehtod instead of just having the code here in `predict`?

Reply:
I added this for `kneighbors_classifier_mg.pyx` since the logic is going to be mostly the same, but a tad different. It turns out, the MG kneighbors_classifier is going to be a tad more complex than I originally thought, so it was removed from this PR. I can revert this design change for now and redo the change in the relevant PR